### PR TITLE
Initial checkin of rna-bam-sort-and-index (samtools) workflow

### LINKED ISSUE #4: Add samtools workflow
For AMP PD RNA data, a samtools workflow was run in Terra on the STAR BAMs.
Add to this repository:

- README describing the setup of the index file
- WDL
- inputs.json

### PR DESCRIPTION
**Pull request by wnojopra**
This PR includes the rna-index workflow which executes samtools on a bam file.

Tested in our test workspace.

Fixes #4 .

### CODE REVIEW

File: rna/rna-index/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# RNAIndex with Samtools
```
Comment (mbookman) [inline rna/rna-index/README.md]: Let's rename this directory to `rna-bam-sort-and-index` and workflow `RNABAMSortAndIndex`

File: rna/rna-index/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+This workflow runs samtools sort and samtools index on the STAR bam file.
```
Comment (mbookman) [inline rna/rna-index/README.md]: s/the STAR bam/a BAM/

File: rna/rna-index/rna-index.inputs.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "##_COMMENT1": "Set your per-sample values: name and BAM path",
+  "RNAIndex.sample_name": "__INSERT_SAMPLE_NAME__",
+  "RNAIndex.star_bam_file":"__INSERT_GCS_PATH_TO_BAM_FILE__",
```
Comment (mbookman) [inline rna/rna-index/rna-index.inputs.json]: Let's change `star_bam_file` to `input_bam_file` throughout.

File: rna/rna-index/rna-index.inputs.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  "##_COMMENT2": "Path to a Docker image containing samtools",
+  "RNAIndex.samtools_docker":"biocontainers/samtools:v1.7.0_cv4",
+
+  "##_COMMENT3": "Would be unusual to take more than 24 hours, so use preemptible",
+  "RNAIndex.preemptible_tries": 5,
+
+  "RNAIndex.num_cpu_cores": 4,
```
Comment (mbookman) [inline rna/rna-index/rna-index.inputs.json]: We ultimately used much more modest VM resources, which went into the 
[Controlling Cloud costs - sample use cases](https://support.terra.bio/hc/en-us/articles/360029772212-Controlling-Cloud-costs-sample-use-cases) Terra article:

> We found that we could index BAMs on a small, single-core VM (n1-standard-1) with a 200 GB disk.
> At preemptible rates of $0.01 per hour for the VM and $0.01 for the disk, these workflows cost pennies
> to complete for each sample.

Please update these inputs to reflect that machine shape.

File: rna/rna-index/rna-index.wdl [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    echo "$(date "+%Y-%m-%d %H:%M:%S") Starting samtools idxstats on new bam"
+    samtools idxstats "${sample_name}.samtools.bam" > idxstats.new.txt
+
+    echo "$(date "+%Y-%m-%d %H:%M:%S") Starting samtools flagstat on old bam"
+    samtools flagstat old_bam_file.bam > flagstat.old.txt
+    echo "$(date "+%Y-%m-%d %H:%M:%S") Starting samtools flagstat on new bam"
+    samtools flagstat "${sample_name}.samtools.bam" > flagstat.new.txt
+
+    if ! diff flagstat.old.txt flagstat.new.txt; then
```
Comment (mbookman) [inline rna/rna-index/rna-index.wdl]: Add a comment:

# Verifying the old and new BAMs produce the same flagstats

**Review by mbookman — CHANGES_REQUESTED** [review]

File: rna/rna-bam-sort-and-index/README.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
-# RNAIndex with Samtools
+# RNABAMSortAndIndex with Samtools
 
-This workflow runs samtools sort and samtools index on the STAR bam file.
+This workflow runs samtools sort and samtools index on a bam file.
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/README.md]: s/bam/BAM

File: rna/rna-bam-sort-and-index/rna-index.inputs.json [hunk 7]
```diff
@@ -1,16 +1,16 @@
 {
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/rna-index.inputs.json]: Please rename the file rna-bam-sort-and-index.inputs.json

File: rna/rna-bam-sort-and-index/rna-index.inputs.json [hunk 8]
```diff
@@ -1,16 +1,16 @@
...
   "##_COMMENT1": "Set your per-sample values: name and BAM path",
   "RNAIndex.sample_name": "__INSERT_SAMPLE_NAME__",
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/rna-index.inputs.json]: The input variable names here should all be changed from RNAIndex to RNABAMSortAndIndex

File: rna/rna-bam-sort-and-index/rna-index.wdl [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ## rna-index.wdl
 ##
-## This workflow runs samtools sort and samtools index on the STAR bam file
+## This workflow runs samtools sort and samtools index on a bam file
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/rna-index.wdl]: s/bam/BAM

File: rna/rna-bam-sort-and-index/rna-index.wdl [hunk 10]
```diff
@@ -26,7 +26,7 @@
 
 workflow RNAIndex {
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/rna-index.wdl]: s/RNAIndex/RNABAMSortAndIndex

File: rna/rna-bam-sort-and-index/rna-index.wdl [hunk 11]
```diff
@@ -38,7 +38,7 @@ workflow RNAIndex {
   call samtools_index {
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/rna-index.wdl]: Let's rename this call/task to `samtools_sort_and_index`

File: rna/rna-bam-sort-and-index/rna-index.wdl [hunk 12]
```diff
@@ -88,6 +88,7 @@ task samtools_index {
     echo "$(date "+%Y-%m-%d %H:%M:%S") Starting samtools flagstat on new bam"
     samtools flagstat "${sample_name}.samtools.bam" > flagstat.new.txt
 
+    # Verifying the old and new BAMs produce the same flagstats
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/rna-index.wdl]: Ah - to be consistent with the rest of the file:

echo "$(date "+%Y-%m-%d %H:%M:%S") Verifying the old and new BAMs produce the same flagstats"

File: rna/rna-bam-sort-and-index/rna-index.wdl [hunk 13]
```diff
@@ -73,10 +73,10 @@ task samtools_index {
     set -o pipefail
 
     echo "$(date "+%Y-%m-%d %H:%M:%S") Creating soft link to old bam"
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/rna-index.wdl]: For the echo's, change "bam" to "BAM"

**Review by mbookman — CHANGES_REQUESTED** [review]

File: rna/rna-bam-sort-and-index/README.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
+# RNABAMSortAndIndex with Samtools
+
+This workflow runs samtools sort and samtools index on a BAM file.
+(http://samtools.sourceforge.net/).
+
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/README.md:5]: It then runs `samtools flagstat` on the old and new files and verifies that they are the same.

File: rna/rna-bam-sort-and-index/README.md [hunk 15]
```diff
@@ -0,0 +1,29 @@
+# RNABAMSortAndIndex with Samtools
+
+This workflow runs samtools sort and samtools index on a BAM file.
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/README.md]: > `samtools sort` and `samtools index`

**Review by mbookman — CHANGES_REQUESTED** [review]
Looks good. A few small things.
Also please import these changes to Terra and verify that the workflow still runs.

**Comment by wnojopra** [comment]
I've re-tested these new changes [here](https://app.terra.bio/#workspaces/fc-amp-pd-alpha/STAR%20testing/job_history/63189eba-ec44-43fb-92f3-60fe140828bd) . Please keep in mind that because we cannot rename a workflow, I had to create a new one. Please disregard the old one.

**Review by mbookman — APPROVED** [review]
LGTM

File: rna/rna-bam-sort-and-index/rna-bam-sort-and-index.inputs.json [hunk 16]
```diff
@@ -0,0 +1,16 @@
+{
+  "##_COMMENT1": "Set your per-sample values: name and BAM path",
+  "RNAIndex.sample_name": "__INSERT_SAMPLE_NAME__",
```
Comment (mbookman) [inline rna/rna-bam-sort-and-index/rna-bam-sort-and-index.inputs.json]: These sample inputs need to be updated to use `RNABAMSortAndIndex`.

**Review by mbookman — CHANGES_REQUESTED** [review]

**Review by mbookman — APPROVED** [review]
LGTM

**Pull request merged**